exercises/number_finder.livemd

```diff
@@ -42,7 +42,20 @@ defmodule NumberFinder do
     iex> NumberFinder.smallest([2, 2, 3, 4, 10, 20, -3])
     -3
   """
+
+  ## Input - Output: inputs a list - outputs a number
+  ## Potential solution: go through the list and 
+  ## keep the smallest number found in the accumulator.
+  ## If the current number is smaller than the current accumulator, update the accumulator
+
   def smallest(number_list) do
+    ## Enum.min(number_list) - This works
+    ## Enum.reduce([1, 2, 3], fn(number, acc) -> if number < acc do number end end)
+    Enum.reduce(number_list, 4, fn number, smallest ->
+      if number < smallest do
```

Comment:
What is your `else` case? 

Remember that if `number < smallest` is false `if` will return `nil`. Then your next accumulator is `nil` and your code will run into problems.

Reply:
Your initial accumulator is `4` here. That works because it's larger than any element in the list. You made a comment about not having a good **neutral** accumulator, so what if your accumulator was just the first element in your list?

Reply:
Thank you! Concepts start to be clarified in my head. Yesterday's class also helped a lot.